src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputAction.java

```diff
@@ -25,19 +29,33 @@
 
     /** JENKINS-37154: number of seconds to block in {@link #loadExecutions} before we give up */
     @SuppressWarnings("FieldMayBeFinal")
-    private static /* not final */ int LOAD_EXECUTIONS_TIMEOUT = Integer.getInteger(InputAction.class.getName() + ".LOAD_EXECUTIONS_TIMEOUT", 10);
+    private static /* not final */ int LOAD_EXECUTIONS_TIMEOUT = Integer.getInteger(InputAction.class.getName() + ".LOAD_EXECUTIONS_TIMEOUT", 60);
 
     private transient List<InputStepExecution> executions = new ArrayList<InputStepExecution>();
     @SuppressFBWarnings(value="IS2_INCONSISTENT_SYNC", justification="CopyOnWriteArrayList")
     private List<String> ids = new CopyOnWriteArrayList<String>();
 
     private transient Run<?,?> run;
 
+    /** ID to InputPromptDefinition */
+    private Map<String,InputPromptDefinition> inputDefinitions;
+
     @Override
     public void onAttached(Run<?, ?> r) {
         this.run = r;
     }
 
+    public synchronized Map<String, InputPromptDefinition> getInputDefinitions() throws InterruptedException, TimeoutException{
+        if (inputDefinitions == null) {
```

Comment:
IIUC this would be for compatibility with builds started before the upgrade, which I guess the existing `@LocalData` test exercises.